### PR TITLE
Master Faskes Service - Bug `is_paginated` & Filtering Updates

### PR DESCRIPTION
**Pull request by rindibudiaramdhan**
## Changes
- Bugfix `is_paginated` was wrong when examine the condition. It must opposite condition from before.
  - Example:
    if `is_paginated` == 1 (true), filter condition must be show data with pagination.

-  Filtering updates:
  - add new filter. `is_faskes`. Value must be boolean. not required. It filter data by faskes (Fasilitas Kesehatan) or non faskes. Faskes in database is include "rumah sakit" (1), "Puskesmas" (2) and "Klinik" (3). other than that is non faskes (4 (Masyarakat Umum) and 5 (Instansi Lainnya)
  - add eager load for village at Master Faskes model.

## Evidence
project: Pikobar Logistik
title:  Master Faskes Service - Bug `is_paginated` & Filtering Updates
participants: @rindibudiaramdhan

### CODE REVIEW

File: app/Http/Controllers/API/v1/MasterFaskesController.php [hunk 1]
```diff
@@ -19,16 +19,23 @@ public function index(GetMasterFaskesRequest $request)
...
                 ->where(function ($query) use ($request) {
                     $query->when($request->has('nama_faskes'), function ($query) use ($request) {
                         $query->where('master_faskes.nama_faskes', 'LIKE', "%{$request->input('nama_faskes')}%");
                     })
                     ->when($request->has('id_tipe_faskes'), function ($query) use ($request) {
                         $query->where('master_faskes.id_tipe_faskes', '=', $request->input('id_tipe_faskes'));
                     })
+                    ->when($request->has('is_faskes'), function ($query) use ($request) {
+                        $query->when($request->input('is_faskes'), function ($query) use ($request) {
+                            $query->whereIn('master_faskes.id_tipe_faskes', [1, 2, 3]);
```
Comment (sandisunandar99) [inline app/Http/Controllers/API/v1/MasterFaskesController.php:34]: CMIWIW .. mangs ,, kalo boleh angka2 constan gini kalo boleh dri pada hanya di tulis angka saja bisa jdi bingung.. kecuali kalo di buat variable atau di kasih komen keterangan supaya tau angka 1 apa , 2 apa , dst..

**Review by setiadijoe — APPROVED** [review]

**Review by ayocodingit — APPROVED** [review]

File: app/Http/Controllers/API/v1/MasterFaskesController.php (hunk 1 above)
Comment (rindibudiaramdhan) [reply app/Http/Controllers/API/v1/MasterFaskesController.php:34]: sebenarnya sudah saya deskripsikan di PR ini.
tapi betul sekali, kebutuhan penjelasan angka ini harus ada di code nya supaya mudah dipahami. siap pak @sandisunandar99

Comment (rindibudiaramdhan) [reply app/Http/Controllers/API/v1/MasterFaskesController.php]: saya buat constants di model `MaskerFaskesType` yang berisi value tsb ya mang @sandisunandar99 

changes nya disini https://github.com/jabardigitalservice/pikobar-logistik-api/pull/341/commits/4c0c070cb428273ebd5bb510266c7974f951fcab

Comment (rindibudiaramdhan) [reply app/Http/Controllers/API/v1/MasterFaskesController.php]: image

Comment (sandisunandar99) [reply app/Http/Controllers/API/v1/MasterFaskesController.php]: ntappss kakaks 👍 😆

**Review by sandisunandar99 — APPROVED** [review]

**Pull request merged**